keps/sig-apps/2890-podGC-delete-all-terminated-pods/README.md

```diff
@@ -0,0 +1,267 @@
+- [x] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [x] (R) Production readiness review completed
+- [x] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This feature allows users to configure whether they want to delete all terminated pods or not.
+
+## Motivation
+
+Currently podGC controller starts cleaning up terminated Pods (with a phase of Succeeded or Failed), when the number of pods exceeds the configured threshold (determined by terminated-pod-gc-threshold in the kube-controller-manager). when user sets terminated-pod-gc-threshold <= 0, the terminated pod garbage collector is disabled. but in some production environment, users do not want to leave any terminated pods. Because of this, it should be opt-in for users to determinate whether they want to delete all terminated pods or not.
+
+### Goals
+
+- a flag that allows users to determinate whether they want to delete all terminated pods or not.
+
+### Non-Goals
+
+- N/A
+
+## Proposal
+
+Add a flag delete-all-terminated-pods in kube-controller-manager to be opt-in for users whether they want to delete all terminated pods or not.
+
+### Risks and Mitigations
+
+When the feature JobTrackingWithFinalizers is disabled, Job controller relies on completed Pods to not be removed in order to track the Job completion status, if user set delete-all-terminated-pods to true, podGC will delete all terminated pods, it will break jobs, 
+the job will never be completed. users should take appropriate care using this feature, when user set delete-all-terminated-pods to true, they should enable the feature JobTrackingWithFinalizers.
+
+## Design Details
+
+This will be launched as an alpha feature first, with feature gate `PodGCDeleteAllTerminatedPods`.
```

Comment:
I think it's not necessary to have a feature flag for a feature that is already controlled by a flag.